sklearn/tests/test_multiclass.py

```diff
@@ -438,36 +465,33 @@ def test_ovo_gridsearch():
 
 
 def test_ovo_ties():
-    # test that ties are broken using the decision function, not defaulting to
-    # the smallest label
+    # Test that ties are broken using the decision function,
+    # not defaulting to the smallest label
     X = np.array([[1, 2], [2, 1], [-2, 1], [-2, -1]])
     y = np.array([2, 0, 1, 2])
     multi_clf = OneVsOneClassifier(Perceptron())
     ovo_prediction = multi_clf.fit(X, y).predict(X)
+    ovo_decision = multi_clf.decision_function(X)
+
+    # Classifiers are in order 0-1, 0-2, 1-2
+    # Use decision_function to compute the votes and the normalized 
+    # sum_of_confidences, which is used to disambiguate when there is a tie in
+    # votes.
+    votes = np.round(ovo_decision)
+    normalized_confidences = ovo_decision - votes
 
-    # recalculate votes to make sure we have a tie
-    predictions = np.vstack([clf.predict(X) for clf in multi_clf.estimators_])
-    scores = np.vstack([clf.decision_function(X)
-                        for clf in multi_clf.estimators_])
-    # classifiers are in order 0-1, 0-2, 1-2
-    # aggregate votes:
-    votes = np.zeros((4, 3))
-    votes[np.arange(4), predictions[0]] += 1
-    votes[np.arange(4), 2 * predictions[1]] += 1
-    votes[np.arange(4), 1 + predictions[2]] += 1
-    # for the first point, there is one vote per class
+    # For the first point, there is one vote per class
     assert_array_equal(votes[0, :], 1)
-    # for the rest, there is no tie and the prediction is the argmax
+    # For the rest, there is no tie and the prediction is the argmax
     assert_array_equal(np.argmax(votes[1:], axis=1), ovo_prediction[1:])
-    # for the tie, the prediction is the class with the highest score
+    # For the tie, the prediction is the class with the highest score
     assert_equal(ovo_prediction[0], 0)
-    # in the zero-one classifier, the score for 0 is greater than the score for
-    # one.
-    assert_greater(scores[0][0], scores[0][1])
-    # score for one is greater than score for zero
-    assert_greater(scores[2, 0] - scores[0, 0], scores[0, 0] + scores[1, 0])
-    # score for one is greater than score for two
-    assert_greater(scores[2, 0] - scores[0, 0], -scores[1, 0] - scores[2, 0])
+
+    # Assert that for the 0th sample with tie in votes
+    # the normalized_confidences is greater for class 0 than class 1
+    assert_greater(
```

Comment:
Either check that also for class two, or maybe
``assert_equal(ovo_prediciton[0], np.argmax(normalized_confidences[0]))``
which is not exactly as strong.